### DISCUSSION TITLE
Acknowledgement and feature requests

### DISCUSSION THREAD

**Discussion by Glaived** [A-tG/VoicemeeterFancyOSD#14] [Ideas]
# Acknowledgement
I wanted to thank you for your work, as soon as I discovered it, it was configured and installed in my environment 👍

# FR Limiter
Add to OSD support for limiters. (Opts: `AlwaysShowElements`, `NeverShowElements`)

# FR Option NeverShowElements
Add an option similar to `AlwaysShowElements`, which will do the opposite, i.e. prevent the display of an element.

- Default value: None
- Possible values: None, Fader, Mono, Mute, EQ, Solo, Buses, Limiter

### Context
<details>
  <summary>Click to expand!</summary>
  
I use macros, and some apply many changes in the Voicemeeter mix.
The fact that these routing changes are all displayed on the OSD is not interesting to me since it is difficult to understand in 5s~ what was the will behind these changes. On the other hand, VB Macro Buttons allows me to understand this better, since I would have my macro in an ON state.

In my case, the `NeverShowElements` option with value `Buses` would be a great help since it would ignore the display of these changes.
</details>

### Example
<details>
  <summary>Click to expand!</summary>
  
If you have set `Limiter` as a value for `NeverShowElements`, when the limit value is changed, the OSD will not be displayed.
And if you change the gain, then the limit while the OSD is still displayed, only the gain change will be visually reflected.
</details>

# FR Macro support
*Nb: I don't know if this is technically possible via the Voicemeeter API.*

Macro support could solve my use case described above. It would no longer be a question of using `NeverShowElements` but of relying on macro support.

When a macro is activated, instead of reflecting the direct changes of the macro, only the state of the macro will be displayed via the OSD, i.e. its name, sub-name and state (and maybe also its `PUSH`/`2P` mode)

### Context
<details>
  <summary>Click to expand!</summary>
  
The difference with the display of VB Macro Buttons and the OSD would be minimal, nevertheless the OSD would allow a quick take into account via a temporary display, without VB Macro Buttons having to remain displayed and without the need to consult ALL the macros to know their status.
</details>

### Exception
Macros with the trigger option enabled will not be taken into account on the OSD because they are automated.

**Comment by A-tG** [comment]
There is no easy way to find what parameters were changed by MacroButtons. API just signals when there was change in any Voicemeeter parameter then program read all parameters and compare to old value to find changes. 

NeverShowElements option is not hard to implement (already done), but adding limiter might take some time, because I will need to figure out what type of Control to use and implement visual template for it
